From 1a70578cb17d63245e91b054fde88f8672a36400 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 9 Oct 2017 20:03:08 +0100
Subject: [PATCH] shim: Work around Steam client's SDL regression with
 XMODIFIERS

The version of SDL shipping with Valve's Steam Client has a regression
that fails to correctly `SDL_InitSubSystem` whenever `XMODIFIERS` is set,
talking to dbus objects with no set address.

Relevant issue: ValveSoftware/steam-for-linux#501

This change will ensure that SDL can initialise correctly on startup,
and no longer make users suffer crashes.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/shim/shim.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/shim/shim.c b/src/shim/shim.c
index 1697d36..8c3055f 100644
--- a/src/shim/shim.c
+++ b/src/shim/shim.c
@@ -71,6 +71,12 @@ int main(int argc, char **argv)
          * usage of dbus by Steam. Help them out */
         setenv("DBUS_FATAL_WARNINGS", "0", 1);
 
+        /* A recent regression is to interpret XMODIFIERS and then fail to
+         * make it beyond `SDL_InitSubSystem` due to `InitIME` failing to
+         * properly use D-BUS ...
+         */
+        unsetenv("XMODIFIERS");
+
         memset(&n_argv, 0, sizeof(char *) * (argc + 2));
 
         /* If we're 64-bit and 32-bit is forced, proxy via linux32 */
-- 
2.14.2

